### DISCUSSION TITLE
How to use mdx with getStaticProps and getStatic paths?

### DISCUSSION THREAD

**Discussion by charlie632** [vercel/next.js#12520] [Help] [answered]
I read this awesome [post](https://www.netlify.com/blog/2020/05/04/building-a-markdown-blog-with-next-9.3-and-netlify/?utm_source=github&utm_medium=blog-cs&utm_campaign=devex) from Cassidy Williams and was motivated to create my own personal blog. However, I'd like to use MDX instead of regular markdown, to have more control on what is the output and be able to use react components.

is there anyway I can implement this approach without having to have a `/pages/blog1.mdx` `/pages/blog2.mdx` and so on?

**Comment by brianlovin** [comment, marked as answer]
Definitely possible! Check out the `blog-starter` [example](https://github.com/zeit/next.js/tree/canary/examples/blog-starter). They aren't using mdx, but the concept is the same: you can have a dedicated directory for your posts (in their case, `_posts`) and use some kind of pseudo [api](https://github.com/zeit/next.js/blob/canary/examples/blog-starter/lib/api.js) to read the posts from the filesystem. See usage on a page [here](https://github.com/zeit/next.js/blob/canary/examples/blog-starter/pages/index.js#L38).
